scene/resources/style_box.cpp

```diff
@@ -32,7 +32,32 @@
 
 #include "scene/main/canvas_item.h"
 
-#include <limits.h>
+void StyleBox::_bind_methods() {
```

Comment:
What's the reason to move the method? We usually put `_bind_methods` next to the constructor, so it was in its right place before.

Reply:
The reason was to reorder the methods so that the order corresponds to the order of declaration in the header file. We are pretty inconsistent with the order in our .cpp-files, so following the header should be done everywhere IMO. I don't see any reason to put it next to the constructor. (Most people will use their IDE to navigate to methods directly anyway, but those who don't might find it more logical to keep the order of the header)

Reply:
Well, we put it next to the constructor pretty consistently. That's where I'd expect it to find in any GUI classes, so that's where it should be here.

We can't keep the order of the header, since header is split into 3 parts for each class, and the definitions are presented as one list, and in most organized classes are grouped by logical groups. Headers should also be grouped by logical groups, but once again those groups are split into up to 3 different categories (private, protected, public). In the CPP file those methods which are logically related should be put next to each other even if in the header they are in completely different places due to the access level.

`_bind_methods` is a meta method, a part of the necessary logical framework for the class to exist. It closely relates to the constructor.

Reply:
For now I moved the methods back. However, I still think we should keep the same order in the header/source file since many developers I know find it surprising when the order doesn't match (which I think outweighs the advantage of logical grouping). That way we could check it via a script in the future (which I think Remi suggested at some point, but I can't remember), because in some files the order is completely off. Maybe we could create an additional `public` section at the top to put the constructors and `_bind_methods` closer together (although I am not a friend of this). But that requires further discussion at another place, maybe the QA channel on RC.